app/src/main/java/org/p2p/wallet/history/ui/detailsbottomsheet/HistoryTransactionDetailsBottomSheetPresenter.kt

```diff
@@ -197,10 +196,13 @@ class HistoryTransactionDetailsBottomSheetPresenter(
         val titleRes: Int
         when (status) {
             HistoryTransactionStatus.COMPLETED -> {
-                val isSend = transaction is RpcHistoryTransaction.Transfer && transaction.isSend
-                val notGreen = transaction is RpcHistoryTransaction.Swap || isSend
                 titleRes = R.string.transaction_details_title_succeeded
-                colorRes = if (notGreen) R.color.text_night else R.color.text_mint
+                colorRes = when {
+                    transaction is RpcHistoryTransaction.Transfer && transaction.isSend ||
+                        transaction is RpcHistoryTransaction.BurnOrMint && transaction.isBurn ||
+                        transaction is RpcHistoryTransaction.Swap -> R.color.text_night
+                    else -> R.color.text_mint
+                }
```

Comment:
```suggestion
val isTransactionTransfer = transaction is RpcHistoryTransaction.Transfer && transaction.isSend
                val isTransactionBurn = transaction is RpcHistoryTransaction.BurnOrMint && transaction.isBurn
                val isTransactionSwap = transaction is RpcHistoryTransaction.Swap
                colorRes = if (isTransactionTransfer || isTransactionBurn || isTransactionSwap) {
                    R.color.text_night
                } else {
                    R.color.text_mint
                }
```

Reply:
why do you need double check RpcHistoryTransaction.Transfer for transaction.isSend? 
i thought Transfer == Send

Reply:
Transfer == Send/Receive